chat_shell/chat_shell/agents/graph_builder.py

```
@@ -156,13 +181,105 @@ def prompt_modifier(state: dict[str, Any]) -> list[BaseMessage]:
+                # This keeps original state.messages unmodified
+                return {"llm_input_messages": compressed_lc_messages}
+
+            # No compression needed, return empty dict (use original messages)
+            return {}
+
+        return pre_model_hook
+
     @trace_sync(
         span_name="agent_builder.build_agent",
         tracer_name="chat_shell.agents",
         extract_attributes=lambda self, *args, **kwargs: {
```

Ruff (0.14.14)

Unused lambda argument: `args`

(ARG005)